flash/vision/style_transfer/model.py

```diff
@@ -0,0 +1,185 @@
+        self.activation = nn.ReLU() if activation else None
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        if self.upsample:
+            input = self.upsample(input)
+
+        output = self.conv(self.pad(input))
+
+        if self.norm:
+            output = self.norm(output)
+        if self.activation:
+            output = self.activation(output)
+
+        return output
+
+
+class Residual(nn.Module):
+
+    def __init__(self, channels: int) -> None:
+        super().__init__()
+        self.conv1 = Conv(channels, channels, kernel_size=3)
+        self.conv2 = Conv(channels, channels, kernel_size=3, activation=False)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        output = self.conv2(self.conv1(input))
+        return output + input
+
+
+class Transformer(nn.Module):
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.encoder = nn.Sequential(
+            Conv(3, 32, kernel_size=9),
+            Conv(32, 64, kernel_size=3, stride=2),
+            Conv(64, 128, kernel_size=3, stride=2),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+            Residual(128),
+        )
+        self.decoder = nn.Sequential(
+            Conv(128, 64, kernel_size=3, stride=2, upsample=True),
+            Conv(64, 32, kernel_size=3, stride=2, upsample=True),
+            Conv(32, 3, kernel_size=9, norm=False, activation=False),
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        return self.decoder(self.encoder(input))
+
+
+class StyleTransfer(Task):
+
+    def __init__(
+        self,
+        style_image: torch.Tensor,
+        model: Optional[nn.Module] = None,
+        multi_layer_encoder: Optional = None,
+        content_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        style_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        metrics: Union[torchmetrics.Metric, Mapping, Sequence, None] = None,
+        learning_rate: float = 1e-3,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+    ):
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+
+        if content_loss is None:
+            content_loss = self.default_content_loss(multi_layer_encoder)
+
+        if style_loss is None:
+            style_loss = self.default_style_loss(multi_layer_encoder)
+
+        self.perceptual_loss = loss.PerceptualLoss(content_loss, style_loss)
+        self.perceptual_loss.set_style_image(style_image)
+
+        self.save_hyperparameters()
```

Comment:
Couldn't find documentation for this. Is this needed? If yes, what does it do?

Reply:
@tchaton @ethanwharris  is this somethings that we could automate inside the main `Task` class ?